exercises/word-search/canonical-data.json

```diff
@@ -0,0 +1,594 @@
+          },
+          "end": {
+            "column": 6,
+            "row": 7
+          }
+        },
+        {
+          "start": {
+            "column": 3,
+            "row": 6
+          },
+          "end": {
+            "column": 6,
+            "row": 3
+          }
+        }
+      ]
+    },
+    {
+      "description": "Should locate words written top right to bottom left",
+      "property": "search",
+      "grid": [
+        "jefblpepre",
+        "camdcimgtc",
+        "oivokprjsm",
+        "pbwasqroua",
+        "rixilelhrs",
+        "wolcqlirpc",
+        "screeaumgr",
+        "alxhpburyi",
+        "jalaycalmp",
+        "clojurermt"
+      ],
+      "wordsToSearchFor": [
+        "clojure",
+        "elixir",
+        "ecmascript",
+        "rust",
+        "java",
+        "lua",
+        "lisp",
+        "ruby"
+      ],
+      "expected": [
+        {
+          "start": {
+            "column": 1,
+            "row": 10
+          },
+          "end": {
+            "column": 7,
+            "row": 10
+          }
+        },
+        {
+          "start": {
+            "column": 6,
+            "row": 5
+          },
+          "end": {
+            "column": 1,
+            "row": 5
+          }
+        },
+        {
+          "start": {
+            "column": 10,
+            "row": 1
+          },
+          "end": {
+            "column": 10,
+            "row": 10
+          }
+        },
+        {
+          "start": {
+            "column": 9,
+            "row": 5
+          },
+          "end": {
+            "column": 9,
+            "row": 2
+          }
+        },
+        {
+          "start": {
+            "column": 1,
+            "row": 1
+          },
+          "end": {
+            "column": 4,
+            "row": 4
+          }
+        },
+        {
+          "start": {
+            "column": 8,
+            "row": 9
+          },
+          "end": {
+            "column": 6,
+            "row": 7
+          }
+        },
+        {
+          "start": {
+            "column": 3,
+            "row": 6
+          },
+          "end": {
+            "column": 6,
+            "row": 3
+          }
+        },
+        {
+          "start": {
+            "column": 8,
+            "row": 6
+          },
+          "end": {
+            "column": 5,
+            "row": 9
+          }
+        }
+      ]
+    },
+    {
+      "description": "Should error if any word is not in the puzzle",
+      "property": "search",
+      "grid": [
+        "jefblpepre",
+        "camdcimgtc",
+        "oivokprjsm",
+        "pbwasqroua",
+        "rixilelhrs",
+        "wolcqlirpc",
+        "screeaumgr",
+        "alxhpburyi",
+        "jalaycalmp",
+        "clojurermt"
+      ],
+      "wordsToSearchFor": [
+        "clojure",
+        "elixir",
+        "ecmascript",
+        "rust",
+        "java",
+        "lua",
+        "lisp",
+        "ruby",
+        "haskell"
+      ],
+      "expected": -1
```

Comment:
If it becomes the case that the output can allow easy identification of which start/end pair corresponds to which word, I didn't find it too necessary to error on the entire output if just a single word is missing.